src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -218,6 +255,11 @@ internal string FullPath
 
         #region Methods
 
+        public bool IsFiltered(string projectFile)
+        {
+            return _solutionFilter == null || _solutionFilter.Contains(projectFile);
+        }
+
```

Comment:
Do you think this method should be more intelligent? In particular, I was wondering if we should check if there's a mismatch between full and relative paths. One extra complication for this is that paths in the .slnf are, by default, relative from the .sln file, which might include some `..`s that would have to be simplified out.

Reply:
I don't understand the question. Can you clarify? I would expect the path constraints on projects in the solution to match the slnf; is that not true?

Reply:
Currently, both the .sln and .slnf use relative paths from the .sln, so there's no problem. As you noted at one point, however, .slnfs are easy to modify, so this would make it resistant to adding an absolute path into the .slnf.

Reply:
I'd rather reject such an slnf than try to tolerate it (unless VS does)